packages/component-library/src/components/table-and-list/sw-data-table/sw-data-table.interactive.stories.ts

```diff
@@ -557,3 +557,47 @@ export const VisualTestEnableRowNumbering: SwDataTableStory = {
     await expect(canvas.getByText("Awesome Concrete Chair")).toBeInTheDocument();
   },
 };
+
+export const EmitOpenDetailsEventOnClickingEdit: SwDataTableStory = {
+  name: "Emit open details event on clicking edit",
+  args: {
+    disableEdit: false,
+  },
+  play: async ({ canvasElement, args }) => {
+    const canvas = within(canvasElement);
+
+    await waitUntilRendered(() => document.querySelectorAll(".sw-skeleton-bar").length === 0);
+
+    const editLink = canvas.getAllByText("Edit")[0];
+
+    await userEvent.click(editLink);
+    
+    await expect(args.openDetails).toHaveBeenCalled();
```

Comment:
It would be cool if we check if the correct event data is emitted (toHaveBeenCalledWith). Otherwise the test would be still green if I would remove the date from the events

Reply:
Gotcha, I oriented at the `VisualTestEnsureSingleSelectionWithoutLoadMore` `itemAdd` event.